Add doc which describes OpenKilda metrics (#5211)
telstra/open-kilda: docs/design/metrics/openkilda-metrics.md

## HA-flow rule metrics
By default, OpenKilda stores raw stats for all HA-flow rules (like it does for [common flows rules](#rule-metrics)) 

* `<prefix>.haflow.raw.bits` - total size of all packets which were processed by an HA-flow rule in bits
* `<prefix>.haflow.raw.bytes` - total size of all packets which were processed by an HA-flow rule in bytes
* `<prefix>.haflow.raw.packets` - total count of packets which were processed by an HA-flow rule

Each of the metrics above have the following tags:
* `ha_flow_id` - String: ID of an HA-flow which a rule belongs, `unknown` if there is no information about an HA-flow
* `flowid` - String: ID of an HA-sub flow which a rule belongs,
             `shared` if rule belongs to both sub flows (rule is a part of HA flow shared path),
             `unknown` if there is no information about a sub flow
* `direction` - String: direction of flow traffic: `forward`, `reverse` or `undefined`
* `switchid` - String: ID of the switch which rule belongs
* `cookie` - String: OpenFlow cookie of the rule
* `inPort` - Number: input port of a rule (if a rule contains in port in OpenFlow match section)
* `outPort` - Number: out port of a rule

> 

Reviewer: need to add tableid tag here

```suggestion
* `tableid` - Number: OpenFlow table ID

```

* `cookie` - String: OpenFlow cookie of a rule which has instruction "go to meter" with the meter id

## Y-flow shared point meter metric
* `<prefix>.yflow.meter.shared.bits` - total size of all packets which were dropped by shared meter of Y-flow meter in bits
* `<prefix>.yflow.meter.shared.bytes` - total size of all packets which were dropped by shared meter of Y-flow in bytes
* `<prefix>.yflow.meter.shared.packets` - total count of packets which were dropped by shared meter of Y-flow

Tags:
* `switchid` - String: ID of a switch which meter belongs
* `miterid` - Number: ID of a meter
* `y_flow_id` - String: ID of a Y-flow which a meter belongs, `unknown` if there is no information about a Y-flow

## Y-flow Y point meter metric
* `<prefix>.yflow.meter.ypoint.bits` - total size of all packets which were dropped by Y-point meter of Y-flow meter in bits
* `<prefix>.yflow.meter.ypoint.bytes` - total size of all packets which were dropped by Y-point meter of Y-flow in bytes
* `<prefix>.yflow.meter.ypoint.packets` - total count of packets which were dropped by Y-point meter of Y-flow

Tags:
* `switchid` - String: ID of a switch which meter belongs

> * `miterid` - Number: ID of a meter

Reviewer: ```suggestion
* `meterid` - Number: ID of a meter
```

* `<prefix>.haflow.meter.packets` - total count of packets which were dropped by HA-flow meter

Tags:
* `switchid` - String: ID of a switch which meter belongs
* `miterid` - Number: ID of a meter
* `ha_flow_id` - String: ID of an HA-flow which a meter belongs, `unknown` if there is no information about an HA-flow
* `flowid` - String: ID of an HA-sub flow which a meter belongs,
             `shared` if rule belongs to both sub flows (meter is a part of HA flow shared path),
             `unknown` if there is no information about a sub flow
* `direction` - String: direction of flow traffic: `forward`, `reverse` or `undefined`
* `cookie` - String: OpenFlow cookie of a rule which has instruction "go to meter" with the meter id

## HA-flow Y point meter metric
* `<prefix>.haflow.meter.ypoint.bits` - total size of all packets which were dropped by Y-point meter of HA-flow meter in bits
* `<prefix>.haflow.meter.ypoint.bytes` - total size of all packets which were dropped by Y-point meter of HA-flow in bytes
* `<prefix>.haflow.meter.ypoint.packets` - total count of packets which were dropped by Y-point meter of HA-flow

Tags:
* `switchid` - String: ID of a switch which meter belongs

> * `miterid` - Number: ID of a meter

Reviewer: ```suggestion
* `meterid` - Number: ID of a meter
```

Tags:
* `flowid` - String: ID of the flow which a rule belongs, `unknown` if there is no information about a flow
* `direction` - String: direction of flow traffic: `forward`, `reverse` or `undefined`
* `is_y_flow_subflow` Boolean: `true` if a rule belongs to a flow, which is a sub flow of some Y-flow, `false` otherwise

### Common flow transit rule metrics
Flow transit rules doesn't have some specific metrics, so kilda just stores [raw](#rule-metrics) metrics.

### Common flow egress rule metrics
* `<prefix>.flow.bits` - total size of all packets which were processed by flow egress rule in bits
* `<prefix>.flow.bytes` - total size of all packets which were processed by flow egress rule in bytes
* `<prefix>.flow.packets` - total count of packets which were processed by flows egress rule

Tags:
* `flowid` - String: ID of the flow which a rule belongs, `unknown` if there is no information about a flow
* `direction` - String: direction of flow traffic: `forward`, `reverse` or `undefined`
* `is_y_flow_subflow` Boolean: `true` if a rule belongs to a flow, which is a sub flow of some Y-flow, `false` otherwise

## Y-flow rule metrics
### Y-flow ingress rule metrics

> * `<prefix>.flow.ingress.bits` - total size of all packets which were processed by Y-flow ingress rule in bits

Reviewer 2: ```suggestion
* `<prefix>.flow.ingress.bits` - total size of all packets which were processed by the Y-flow ingress rule in bits
```

> * `<prefix>.flow.ingress.bytes` - total size of all packets which were processed by Y-flow ingress rule in bytes

Reviewer 2: ```suggestion
* `<prefix>.flow.ingress.bytes` - total size of all packets which were processed by the Y-flow ingress rule in bytes
```

> * `<prefix>.flow.ingress.packets` - total count of packets which were processed by Y-flows ingress rule

Reviewer 2: ```suggestion
* `<prefix>.flow.ingress.packets` - total count of packets which were processed by the Y-flow's ingress rule
```

* `<prefix>.yflow.ypoint.bytes` - total size of all packets which were processed by Y-flow rule in Y point in bytes
* `<prefix>.yflow.ypoint.packets` - total count of packets which were processed by Y-flow rule in Y point

Tags:
* `y_flow_id` - String: ID of a Y-flow which a rule belongs, `unknown` if there is no information about a Y-flow
* `flowid` - String: ID of a sub flow which a rule belongs, `unknown` if there is no information about a flow
* `direction` - String: direction of flow traffic: `forward`, `reverse` or `undefined`
* `is_y_flow_subflow` Boolean: `true` as a sub flow belongs to a Y-flow

## HA-flow rule metrics
By default, OpenKilda stores raw stats for all HA-flow rules (like it does for [common flows rules](#rule-metrics)) 

* `<prefix>.haflow.raw.bits` - total size of all packets which were processed by an HA-flow rule in bits
* `<prefix>.haflow.raw.bytes` - total size of all packets which were processed by an HA-flow rule in bytes
* `<prefix>.haflow.raw.packets` - total count of packets which were processed by an HA-flow rule

Each of the metrics above have the following tags:

> * `ha_flow_id` - String: ID of an HA-flow which a rule belongs, `unknown` if there is no information about an HA-flow

Author: We already have `y_flow_id` so we decided to use `ha_flow_id`

* `y_flow_id` - String: ID of a Y-flow which a rule belongs, `unknown` if there is no information about a Y-flow
* `flowid` - String: ID of a sub flow which a rule belongs, `unknown` if there is no information about a flow
* `direction` - String: direction of flow traffic: `forward`, `reverse` or `undefined`
* `is_y_flow_subflow` Boolean: `true` as a sub flow belongs to a Y-flow

### Y-flow Y point rule metrics
* `<prefix>.yFlow.yPoint.bits` - total size of all packets which were processed by Y-flow rule in Y point in bits
* `<prefix>.yFlow.yPoint.bytes` - total size of all packets which were processed by Y-flow rule in Y point in bytes
* `<prefix>.yFlow.yPoint.packets` - total count of packets which were processed by Y-flow rule in Y point

Tags:
* `y_flow_id` - String: ID of a Y-flow which a rule belongs, `unknown` if there is no information about a Y-flow
* `flowid` - String: ID of a sub flow which a rule belongs, `unknown` if there is no information about a flow
* `direction` - String: direction of flow traffic: `forward`, `reverse` or `undefined`
* `is_y_flow_subflow` Boolean: `true` as a sub flow belongs to a Y-flow

## HA-flow rule metrics
### HA-flow ingress rule metrics

> * `<prefix>.haflow.ingress.bits` - total size of all packets which were processed by HA-flow ingress rule in bits

Reviewer: here <prefix>.haflow.ingress.bits
should we use "haflow" instead of "haFlow" ?

just for comparison, for y-flow metrics, we use:
`<prefix>.yFlow.yPoint.bytes`

Author: lets discuss the name online

* `<prefix>.haflow.ingress.bytes` - total size of all packets which were processed by HA-flow ingress rule in bytes
* `<prefix>.haflow.ingress.packets` - total count of packets which were processed by HA-flows ingress rule

Tags:
* `ha_flow_id` - String: ID of an HA-flow which a rule belongs, `unknown` if there is no information about an HA-flow
* `flowid` - String: ID of an HA-sub flow which a rule belongs,
             `shared` if rule belongs wo both sub flows (rule is a part of HA flow shared path),
             `unknown` if there is no information about a sub flow
* `direction` - String: direction of flow traffic: `forward`, `reverse` or `undefined`

### HA-flow transit rule metrics
* `<prefix>.haflow.transit.bits` - total size of all packets which were processed by an HA-flow transit rule in bits
* `<prefix>.haflow.transit.bytes` - total size of all packets which were processed by an HA-flow transit rule in bytes

> * `<prefix>.haflow.transitraw.packets` - total count of packets which were processed by an HA-flow transit rule

Author: fixed

Each of the metrics above have the following tags:
* `ha_flow_id` - String: ID of an HA-flow which a rule belongs, `unknown` if there is no information about an HA-flow
* `flowid` - String: ID of an HA-sub flow which a rule belongs,

>              `shared` if rule belongs wo both sub flows (rule is a part of HA flow shared path),

Reviewer: ```suggestion
             `shared` if rule belongs to both sub flows (rule is a part of HA flow shared path),
```

Author: fixed

* `flowid` - String: ID of an HA-sub flow which a rule belongs,
             `shared` if rule belongs wo both sub flows (rule is a part of HA flow shared path),
             `unknown` if there is no information about a sub flow
* `direction` - String: direction of flow traffic: `forward`, `reverse` or `undefined`

REMOVE???
### HA-flow shared point rule metrics
* `<prefix>.haflow.shared.bits` - total size of all packets which were processed by HA-flow rule in shared point in bits
* `<prefix>.haflow.shared.bytes` - total size of all packets which were processed by HA-flow rule in shared point in bytes
* `<prefix>.haflow.shared.packets` - total count of packets which were processed by HA-flow rule in shared point

Tags:
* `ha_flow_id` - String: ID of an HA-flow which a rule belongs, `unknown` if there is no information about an HA-flow
* `flowid` - String: ID of an HA-sub flow which a rule belongs,
             `shared` if rule belongs wo both sub flows (rule is a part of HA flow shared path),
             `unknown` if there is no information about a sub flow
* `direction` - String: direction of flow traffic: `forward`, `reverse` or `undefined`

### HA-flow Y point rule metrics

> * `<prefix>.halow.yPoint.bits` - total size of all packets which were processed by HA-flow rule in Y point in bits

Reviewer 3: 'halow' typo

Author: fixed

* `<prefix>.haflow.yPoint.bytes` - total size of all packets which were processed by HA-flow rule in Y point in bytes
* `<prefix>.haflow.yPoint.packets` - total count of packets which were processed by HA-flow rule in Y point

* `ha_flow_id` - String: ID of an HA-flow which a rule belongs, `unknown` if there is no information about an HA-flow
* `flowid` - String: ID of an HA-sub flow which a rule belongs,
             `shared` if rule belongs wo both sub flows (rule is a part of HA flow shared path),
             `unknown` if there is no information about a sub flow
* `direction` - String: direction of flow traffic: `forward`, `reverse` or `undefined`

## Meter metrics

> > **NOTE:** All meter metrics will have value greater than zero only if traffix rate will exceed meter rate

Author: fixed
